develop/components/pages/DropdownPage.tsx

```diff
@@ -1,123 +1,130 @@
-            dropDownList3: [
-                { value: "1", label: "Paraguay" },
-                { value: "2", label: "Dominican Republic" },
-                { value: "3", label: "Mongolia" },
-                { value: "4", label: "Montserrat" },
-                { value: "5", label: "Thailand" },
-                { value: "6", label: "Japan" },
-                { value: "7", label: "Saint Vincent and the Grenadines" },
-            ],
-            dropDownList3Selected: null,
-            dropDownList4: [
-                { value: "1", label: "Sierra Leone" },
-                { value: "2", label: "Malawi" },
-                { value: "3", label: "Marshall Islands" },
-                { value: "4", label: "Latvia" },
-                { value: "5", label: "Slovenia" },
-                { value: "6", label: "Argentina" },
-                { value: "7", label: "Solomon Islands" },
-            ],
-            dropDownList4Selected: null
-        };
+const DropdownPage: React.FunctionComponent = () => {
+    const [dropDownList1Selected, setDropDownList1Selected] = React.useState<DropDownItem>(null);
+    const [dropDownList2Selected, setDropDownList2Selected] = React.useState<Array<DropDownItem>>(null);
+    const [dropDownList3Selected, setDropDownList3Selected] = React.useState<DropDownItem>(null);
+    const [dropDownList4Selected, setDropDownList4Selected] = React.useState<DropDownItem>(null);
+    const [disabled, setDisabled] = React.useState<boolean>(false);
 
-        this.onChangeDropdown = this.onChangeDropdown.bind(this);
-    }
+    return (
+        <div className="route-template container">
+            <div className="info-holder">
 
-    onChangeDropdown(value: DropDownItem | Array<DropDownItem> | string, name: string) {
-        this.setState({ [name]: value });
-    }
-
-    render() {
-        const mode = getParameterByName(this.props.location.search, "mode");
-        return (
-            <div className={"route-template " + ((mode === "dl" || mode === "DL") ? "brief" : "")}>
-                <div className="info-holder">
+                <div className="info">
+                    <div className="md-file">
+                        <Highlight innerHTML={true}>{docMD}</Highlight>
+                    </div>
+                </div>
 
-                    <div className="info">
-                        <div className="md-file">
-                            <Highlight innerHTML={true}>{docMD}</Highlight>
-                        </div>
+                <div className="info">
+                    <h2>Output</h2>
+                    <p>Here is the basic one:</p>
+                    <div className="result">
+                        <DropDown
+                            label="Single select"
+                            list={dropDownList1}
+                            selectedValue={dropDownList1Selected}
+                            onChange={(value: DropDownItem) => setDropDownList1Selected(value)}
```

Comment:
```suggestion
                            onChange={setDropDownList1Selected}
```

Reply:
There are two types of the value being returned here:
```typescript
selectedValue: DropDownItem | Array<DropDownItem>
```
If I do what you suggested, typescript will complain. That's why I'm selectively choosing the first variation `DropDownItem` and the next example I'm selecting `Array<DropDownItem>` as the type of the value being returned.

Reply:
alright